test/integration/targets/aws_ses_identity_dkim/tasks/main.yml

```diff
@@ -0,0 +1,206 @@
+---
+# tasks file for aws_ses_identity_dkim test
+
+- name: set up aws connection info
+  set_fact:
+    aws_connection_info: &aws_connection_info
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token }}"
+      region: "{{ aws_region }}"
+  no_log: yes
+
+- block:
+
+    ## prepare SES idenitiy to run tests on
+
+    - name: Create SES identity for running DKIM module tests on (domain identity)
+      aws_ses_identity:
+        identity: "{{ dkim_identity_domain }}"
+        state: present
+        <<: *aws_connection_info
+      register: ses_domain_identity
+
+    - name: Create SES identity for running DKIM module tests on (email identity)
+      aws_ses_identity:
+        identity: "{{ dkim_identity_email }}"
+        state: present
+        <<: *aws_connection_info
+      register: ses_email_identity
+
+    ## DKIM check mode enable test (nothing should change, dry-run)
+
+    - name: Enable DKIM verification on email identity (check-mode)
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_email }}"
+        state: enabled
+        <<: *aws_connection_info
+      check_mode: True
+      register: email_result
+
+    - name: assert results report being unchanged (check-mode)
+      assert:
+        that:
+          - email_result.changed == False
```

Comment:
I don't think this is correct (I should have renamed the task). In general check mode should return "as if it made the change". Doing the actually attributes for this tends to be hard but at least the changed flag should represent whether a change whould have been made.

So if the task would have made a change if it were not in check mode, it should return True from changed even though it didn't actually make a change.

That way you can use check mode to see  what changes would be made.

Reply:
Ok, then I got the check mode purpose slightly wrong.